cpp/tests/io/experimental/hybrid_scan_common.hpp

```diff
@@ -139,25 +148,56 @@ auto create_parquet_with_stats(
 
   auto col2 = [&]() {
     if constexpr (IsConstantStrings) {
-      return constant_strings(col2_value);  // constant stringified value
+      return constant_strings(str_col_value);  // constant stringified value
     } else {
       return testdata::ascending<cudf::string_view>();  // ascending strings
     }
   }();
 
-  auto expected = table_view{{col0, col1, col2}};
-  auto table    = cudf::concatenate(std::vector<table_view>(NumTableConcats, expected));
-  expected      = table->view();
+  // Output table view
+  auto output = table_view{{col0, col1, col2}};
+
+  // Do we need to create and add nullmasks to the columns?
+  std::vector<std::unique_ptr<cudf::column>> columns;
+  if constexpr (IsNullable) {
+    std::mt19937 gen(0xc0ffee);
+    std::bernoulli_distribution bn(0.7f);
+    auto valids =
+      cudf::detail::make_counting_transform_iterator(0, [&](int index) { return bn(gen); });
+    auto const num_rows = static_cast<cudf::column_view>(col0).size();
+
+    columns.emplace_back(col0.release());
+    auto [nullmask, nullcount] = cudf::test::detail::make_null_mask(valids, valids + num_rows);
+    columns.back()->set_null_mask(std::move(nullmask), nullcount);
+
+    columns.emplace_back(col1.release());
+    std::tie(nullmask, nullcount) =
+      cudf::test::detail::make_null_mask(valids + num_rows, valids + 2 * num_rows);
+    columns.back()->set_null_mask(std::move(nullmask), nullcount);
+
+    columns.emplace_back(col2.release());
+    std::tie(nullmask, nullcount) =
+      cudf::test::detail::make_null_mask(valids + 2 * num_rows, valids + 3 * num_rows);
```

Comment:
Doesn't really matter what absolute iterator offsets we provide here as long as the distance(start, end) == num_rows